src/js/entry.js

```diff
@@ -208,3 +219,24 @@ export function rootApp() {
         );
     }
 }
+
+export function noAccess() {
+    const { store } = spinUpStore();
+    const currPath = location.pathname;
+    const app = currPath.split('/').pop();
+    const userEntitlements = window.insights.chrome.auth.getUserEntitlements();
+    const apps = { insights: userEntitlements.insights.is_entitled };
+    //Only restrict this in settings/applications for now
+    if (currPath.includes('/settings/applications') && app in apps) {
+        const grantAccess = apps[app];
+        if (!grantAccess) {
+            document.getElementById('root').style.display = 'none';
+            render(
+                <Provider store={ store }>
+                    <NoAccess appName={ app } />
+                </Provider>,
+                document.querySelector('.pf-c-page__no-access')
```

Comment:
You can replace the entire `#no-access` container, that way you will have full access to the screen from within `NoAccess` app.